File: Benchmarking/30 Euler/euler14/euler14.cpp
//
//  main.cpp
//  Euler14
//
//  Created by Sonni on 20/05/16.
//  Copyright © 2016 Sonni Jensen. All rights reserved.
//

#include <iostream>

void seq(unsigned int n)
{
    unsigned int max = 0;
    unsigned int max_number = 0;
    for (unsigned int i = 1; i <= n; i++)
    {
        unsigned int count_seq = 1;
        unsigned int temp = i;
        unsigned int tmpI = i;
        while (tmpI != 1)
        {
            
            if (tmpI % 2 == 0)
                tmpI /= 2;
            else
                tmpI = 3 * tmpI + 1;
            
            count_seq++;
        }
        if (count_seq > max)
        {
            max = count_seq;
            max_number = temp;
        }
    }
    
    return max_number;
}

int main(int argc, const char * argv[]) {
    seq(atoi(argv[1]));
    return 0;
}

File: Project-Euler/Stats/euler14/euler14.py
import sys

limit = int(sys.argv[1])
def seq(n):
  max = 0
  maxNumber = 0
  for i in range(1, n+1):
    countSeq = 1
    temp = i
    current = i
    while current != 1:
      if current % 2 == 0:
        current = current / 2
        countSeq += 1
      else:
        current = 3 * current + 1
        countSeq += 1
    if countSeq > max:
      max = countSeq
      maxNumber = temp
  return maxNumber

print seq(limit)
